docs/ccv1_image_security_design.md

```diff
@@ -0,0 +1,403 @@
+- Match the default scope under the member in the transport. for example, the "" scope of the transport "dir".
+- Match the global default scope, such as the "default" field in the first line of the above example. We suggest that this policy must be configured to reject for security.
+
+After the matching is successful, the policy set under the corresponding scope is executed. In the design of confidential containers V1, only three policy types are currently supported now: "reject", "insecureAcceptAnything" and "signedBy". The former two is as the name implies, and the "signedBy" policy requires that there must exist at least one signature of this image can be verified by the public key ring provided by `keyPath` or `keyData` filed. In CCv1, he format of "signedBy" policy is defined as follows:
+
+```json
+{
+    "type":    "signedBy",
+    "keyType": "GPGKeys", /* The only currently supported value */
+    "keyPath": "/path/to/local/keyring/file", /* specifies the GPG Keyring that contains one or more public keys. */
+    /* Use one of the keyPath and keyData fields is enough */
+    "keyData": "base64_encoded_keyring_data",
+}
+```
+
+For the example of the policy file given above, the meaning of its security policy is as follows:
+
+1. For the image from the local file system, allowed to pull and run no matter there is a valid signature or not.
+
+2. For the image from the "docker.io/my_private_registry" registry, there must exist at least one signature that can be verified by the keys in keyData.
+
+3. For the image from the "registry.access.redhat.com", there must exist at least one signature that can be verified by the keys in `/etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release`.
+
+4. Reject all other images.
+
+The above example provides a secure and robust policy configuration, but the policy file still provides sufficient configuration flexibility for weak security requirements. For example, if you need to completely disable the container image signature verification function, policy file can be:
+
+```json
+{
+  "default": [
+    {
+      "type": "insecureAcceptAnything"
+    }
+  ],
+}
+```
+
+We must remind that this configuration is very insecure. It will pull all unsigned container images. Or if you want to unconditionally accept an image from a specific registry, you just need to change the policy of the member matching the registry under the corresponding scope to `insecureAcceptAnything`.
+
+### sigstore config
+
+The configuration file that specifies the container image sigstore is specified as all `.yaml` files in /etc/containers/registries.d directory. After the policy.json allowlist check is completed, the entries in the sigstore configuration file will claim where the container image is located, and the signature claim file of the container image will be downloaded from the specified URL (a webserver or a local directory). The sample sigstore configuration file matching the sample policy.json file in the above is as follows:
```

Comment:
Since you wrote "all `.yaml` files", I suspect the sample file below could also be split into three YAML files, right?

Is there any reference for this `yaml` format? Would it accept `oci` or `tarball` like the `policy.json` file?

Reply:
Is there any reference for this `yaml` format? Would it accept `oci` or `tarball` like the `policy.json` file?

Reply:
What are the accepted transports for URLs? I see `file` and https`, but would `http` be accepted?

Reply:
> Since you wrote "all `.yaml` files", I suspect the sample file below could also be split into three YAML files, right?

Yes, when trying to parse sigstore config file, the contents of all yaml files in this directory will be merged.